app/src/main/java/com/ntnu/wip/nabl/Network/FirestoreImpl/FireStoreClient.java

```diff
@@ -271,6 +278,87 @@ public void getLogEntries(String uid, String cid, String pid, long startMillis,
         query.get();
     }
 
+    /**
+     * The goal of this monstrosity, is to fetch the following hierachy
+     * - Company
+     *   - Clients
+     *     - WorkDays
+     *   - Projects
+     *     - WorkDays
+     * This should be returned as a list of companyContainer
+     * @param companies
+     */
+    public void getLogEntriesByCompany(String correlationId, List<Company> companies) {
```

Comment:
This function should in all honesty be refactored.

Reply:
It should. However, the extraction of data is quite complex. When fiddeling with the code you need to have will and time. The rabbit hole is deep